### ISSUE TITLE
Add support for custom heading ID's

### ISSUE THREAD

**Issue by kimburgess** [PlaceOS/docs-site#19]
Implement a parser that allows auto-generated heading anchors to be overridden based on a [commonly used markdown extensions syntax](https://www.markdownguide.org/extended-syntax/#heading-ids).

```markdown
### My Great Heading {#custom-id}
```
Should render as
```html
<h3 id="custom-id">My Great Heading</h3>
```

Background discussion and reasoning for introduction in https://github.com/PlaceOS/docs/issues/45.

**Comment by mghill** [comment]
Would this totally override the heading component of the URL? Or would both be live?
 i.e. the section under "My Great Heading" is now reachable at `.../filename#custom-id` but is it still also reachable at `.../filename#my-great-heading` or not? Not trying to impose a requirement, but just getting a picture of how this works.

**Comment by kimburgess** [comment]
Each element can only have a single id, so this overrides.

**Comment by kimburgess** [comment]
Support added into upstream via facebook/docusaurus#4222.

**Comment by kimburgess** [comment]
Resolved via #21.
